### PR TITLE
Update Provincial Drugs Loader job to use Pharmacy Assessment file for AB#15595

### PR DESCRIPTION
**Pull request by BrianMaki**
# Implements [AB#15595](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/15595)

## Description

Update Provincial Drugs Loader job to use Pharmacy Assessment file for [AB#15595](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/15595).

**Database changes:**

- Add PHAR-ASSMT to ProgamTypeCode table
- Add PHAR-ASSMT to ProgramType file
- Add seed data step to GatewayDbContext - creates a migration

**PharmacyAssessment Mapper:**

- Created model for pharmacy assessment
- Created parser for pharmacy assessment
- Created mapper for pharmacy assessment

**Add pharmacy assessment mapper to DI:**

- JobScheduler program
- DbMaintainer program

**Updated application json files for all environments:**

- Add URL for pharmacy assessment file
- Add target folder for pharmacy assessment file
- Add program code for pharmacy assessment file

**BaseDrugApp changes:**

- Change access level on methods so that they can be accessed or overridden from  child class
- Updated linq call based on Rider suggestion

**FedDrugDBApp changes:**

- Minor accessibility change based on changes in BaseDrugApp - Rider suggestion

**BCPProvDBApp changes:**

- Implemented override so that both Pharma Care Drug and Pharmacy Assessment files will be looked at

## Testing

- [ ] Unit Tests Updated
- [ ] Functional Tests Updated
- [x] Not Required

## UI Changes

No

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

File: Apps/DBMaintainer/Mappers/Converters/BooleanConverter.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        {
+            if (ConvertToBool(text, out bool result))
+            {
+                return result;
+            }
+
+            return null;
+        }
+
+        private static bool ConvertToBool(string text, out bool result)
```
[the remark this answers is not in the document]
Comment (BrianMaki) [reply Apps/DBMaintainer/Mappers/Converters/BooleanConverter.cs]: Changed

File: Apps/DBMaintainer/Mappers/Converters/BooleanConverter.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+            if (!string.IsNullOrWhiteSpace(text))
+            {
+                string boolValue = text.ToUpperInvariant();
+
+                result = boolValue switch
+                {
+                    "YES" or "TRUE" => true,
+                    "NO" or "FALSE" => false,
+                    _ => false,
```
[the remark this answers is not in the document]
Comment (BrianMaki) [reply Apps/DBMaintainer/Mappers/Converters/BooleanConverter.cs:50]: Changed.

[the remark this answers is not in the document]
Comment (MikeLyttle) [reply Apps/DBMaintainer/Mappers/Converters/BooleanConverter.cs:50]: I meant the _ branch should not set result to false, it should cause the method to return false.

[the remark this answers is not in the document]
Comment (BrianMaki) [reply Apps/DBMaintainer/Mappers/Converters/BooleanConverter.cs]: Changed.

**Review by MikeLyttle — APPROVED** [review]

**Review by sslaws — APPROVED** [review]

**Pull request merged**